mithril-aggregator/src/services/prover.rs

```diff
@@ -245,47 +468,136 @@ mod tests {
 
     #[tokio::test]
     async fn compute_proof_for_one_set_of_three_known_transactions_and_two_unknowns() {
-        let (transaction_hashes, transactions) = generate_transactions(5);
-        let prover = build_prover(|retriever_mock| {
-            // The last two are not in the "store"
-            let transactions = transactions[0..=2].to_vec();
-            let transactions_by_hashes_res = transactions.clone();
-            retriever_mock
-                .expect_get_by_hashes()
-                .with(eq(transaction_hashes.clone()))
-                .return_once(move |_| Ok(transactions_by_hashes_res));
-            retriever_mock
-                .expect_get_up_to()
-                .with(eq(fake_data::beacon()))
-                .return_once(move |_| Ok(transactions));
-        });
+        let total_block_ranges = 5;
+        let total_transactions_per_block_range = 3;
+        let transactions = test_data::generate_transactions(
+            total_block_ranges,
+            total_transactions_per_block_range,
+        );
+        let transactions_to_prove =
+            test_data::filter_transactions_for_indices(&[1, 2, 4], &transactions);
+        let transaction_hashes_unknown =
+            vec!["tx-unknown-123".to_string(), "tx-unknown-456".to_string()];
+        let mut test_data = test_data::build_test_data(&transactions_to_prove, &transactions);
+        let transaction_hashes_known = test_data.transaction_hashes_to_prove.clone();
+        test_data.transaction_hashes_to_prove = test_data
+            .transaction_hashes_to_prove
+            .clone()
+            .into_iter()
+            .chain(transaction_hashes_unknown.into_iter())
+            .collect::<Vec<_>>();
```

Comment:
Using `concat` is a bit simpler:
```suggestion
        test_data.transaction_hashes_to_prove: Vec<TransactionHash> = [
           transaction_hashes_to_prove.clone(),
           transaction_hashes_unknown,
         ]
         .concat();
```

Reply:
Done